src/main/groovy/edu/oregonstate/mist/personsapi/PersonsResource.groovy

```diff
@@ -971,4 +972,127 @@ class PersonsResource extends Resource {
             notFound().build()
         }
     }
+
+    // Validate new phone object
+    private List<Error> newPhoneErrors(PhoneObject phone) {
+        List<Error> errors = []
+        Closure addBadRequest = { String message ->
+            errors.add(Error.badRequest(message))
+        }
+
+        [
+            [true, "addressType", 2,
+             { String addressType -> bannerPersonsReadDAO.isValidAddressType(addressType) }
+            ],
+            [true, "phoneType", 2,
+             { String phoneType -> bannerPersonsReadDAO.isValidPhoneType(phoneType) }
+            ],
+            [false, "areaCode", 3, null],
+            [false, "phoneNumber", 7, null],
+            [false, "phoneExtension", 4, null],
+            [false, "primaryIndicator", 5, null],
+        ].each {
+            Boolean isRequired = it.get(0)
+            String fieldName = it.get(1)
+            String fieldValue = phone[fieldName]
+            Integer length = it.get(2)
+            Closure validateFunction = it.get(3)
+
+            // Check if required field is missing
+            if (isRequired && !fieldValue) {
+                addBadRequest("Required field $fieldName is missing or null.")
+            }
+
+            // Check if input field is over the buffer size
+            if (fieldValue?.length() > length) {
+                addBadRequest("$fieldName can't be more than $length characters.")
+            }
+
+            // Check if input field is valid
+            if (fieldValue && validateFunction && !validateFunction(fieldValue)) {
+                addBadRequest("$fieldName is not valid.")
+            }
+        }
+
+        errors
+    }
```

Comment:
I tried this data packet and the data was inserted without an issue:

```
{
    "data": {
        "type": "phones",
        "attributes": {
            "addressType": "RE",
            "phoneType": "CM",
            "phoneNumber": "ABCDEFG",
            "primaryIndicator": "true",
            "phoneExtension": "BAR",
            "areaCode": "FOO"
        }
    }
}
``` 

I guess we should make sure these phone strings are numeric strings? (and add the pattern to swagger)

Reply:
Adding the pattern in swagger doesn't do anything. There's probably something I'm missing. I added this to the existing parameter validation in groovy for the mean time.

Reply:
Right, the swagger file in the old skeleton is just for documentation, we need to implement the validation logic by ourselves. However, it's still worthy to add to the swagger so that our clients can easily pick it up.